vignettes/mobi-projects.Rmd

```diff
@@ -0,0 +1,266 @@
+simulationsInProject <- myProject$simulationNames
+
+# EXPECTED BEHAVIOR
+simulationsInProject <- c("Thyroid_QST_Human", "Thyroid_QST_Phenobarbital")
+
+print(simulationsInProject)
+```
+
+get the list of parameter identifications:
+
+```{r getPInNames, execute = FALSE}
+PIs <- myProject$ParameterIdentificationNames
+
+# EXPECTED BEHAVIOR
+PIs <- c()
+
+print(PIs)
+```
+
+get observed data from the project as `DataSet` objects:
+
+```{r getObservedData}
+obsData <- myProject$getObservedData()
+
+print(names(obsData))
+```
+
+get the list of individuals and expression profiles from the project:
+
+```{r getIndividualsExpressions}
+individualsInProject <- myProject$Individuals
+expProfilesInProject <- myProject$ExpressionProfiles
+
+# EXPECTED BEHAVIOR
+individualsInProject <- c("Human", "Rat")
+expProfilesInProject <- c("UDPGT1|Human|Healthy",
+                          "DIO1|Human|Healthy",
+                          "DIO3|Human|Healthy",
+                          "UDPGT2|Human|Healthy",
+                          "UGT1A1|Rat|Healthy",
+                          "PB-LiverBindingPartner|Human|Healthy")
+```
+
+
+get the modules present in the project:
+
+```{r getModulesNames}
+modulesInProject <- myProject$Modules
+
+# EXPECTED BEHAVIOR
+names(modulesInProject) <- c("Endogenous_TH",
+                             "Phenobarbital_Extension",
+                             "Phenobarbital_PBPK",
+                             "Pituitary",
+                             "Rat physiology",
+                             "TH_activeTransports",
+                             "TH_plasma_binding",
+                             "Thyroid",
+                             "Thyroid_QST")
+
+```
+
+and get the building blocks defined for the modules:
+
+```{r getBuildingBlocksFromModule}
+bbs <- modulesInProject[["Thyroid"]]$getBuildingBlocks()
+
+for (bbName in names(bbs)){
+  print(name, ": ", bbs[[name]]$type)
+}
+
+# Get all parameter values BBs
+paramValuesBBs <- modulesInProject[["Thyroid"]]$getBuildingBlocks(type = "Parameter Values")
+```
+
+We can retrieve simulations from the project:
+
+```{r getSimulation}
+simulation <- myProject$getSimulation[[simulationsInProject[[1]]]]
+```
+
+and the observed data as `DataSet` objects:
+
+```{r getObservedData}
+obsData <- myProject$getObservedData()
+```
+
+## Creating simulations
+
+The user can create new simulation from the modules available in the project. In this example,
+we create a simulation for the rat species from the modules "Thyroid_QST", "Rat physiology", "Thyroid", "Pituitary", "Endogenous_TH", "TH_activeTransports", and "TH_plasma_binding". We will
+select the expression profiles "UDPGT1|Human|Healthy", "DIO1|Human|Healthy", "DIO3|Human|Healthy",
+"UDPGT2|Human|Healthy", and specify the parameter values building block "Rat" from the 
+module "Thyroid":
+
+```{r createSimulationFromProject}
+sim1 <- myProject$createSimulation(simulationName = "Thyroid_QST_rat", 
```

Comment:
@benjaperez1983 and I implemented an integration test this way as well and as a test it works.

With this workflow, you can create a simulation only if the project already has all the modules, and you cannot use any other modules from another source.

However, I think a better idea might be to have the createSimulation work with `Modules` and not have a built-in dependency on the project....

The workflow would be to get the modules and bb's from the project (or file system, or github) then create the simulation from the modules and bb's independently of the project.

Reply:
ok, reading the comments below, I see this is planned, but I think doing it by supplying the strings is going to be a sort of 'limited' version of the code below. I don't see a need for it.

Reply:
Sure, you can also hide whatever abstractions in R code as well.